File: PersonalMac/PersonalMac/Services/SettingsService.swift
import Foundation

actor SettingsService {
    private let settingsDirectory: URL
    private let settingsFile: URL

    init() {
        let appSupport = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask).first!
        settingsDirectory = appSupport.appendingPathComponent("PersonalMac", isDirectory: true)
        settingsFile = settingsDirectory.appendingPathComponent("settings.json")
    }

    func getApiKey() -> String? {
        guard FileManager.default.fileExists(atPath: settingsFile.path) else { return nil }
        guard let data = try? Data(contentsOf: settingsFile),
              let json = try? JSONSerialization.jsonObject(with: data) as? [String: String]
        else { return nil }
        return json["ApiKey"]
    }

    func saveApiKey(_ apiKey: String) throws {
        try FileManager.default.createDirectory(at: settingsDirectory, withIntermediateDirectories: true)
        let json = ["ApiKey": apiKey]
        let data = try JSONSerialization.data(withJSONObject: json, options: .prettyPrinted)
        try data.write(to: settingsFile, options: .atomic)
    }
}

File: PersonalElectron/electron/services/SettingsService.ts
import Store from 'electron-store'

interface SettingsSchema {
  apiKey: string
}

const store = new Store<SettingsSchema>({
  defaults: {
    apiKey: ''
  }
})

export class SettingsService {
  getApiKey(): string {
    return store.get('apiKey', '')
  }

  saveApiKey(key: string): void {
    store.set('apiKey', key)
  }

  hasApiKey(): boolean {
    return !!store.get('apiKey', '')
  }
}
